src/app/server/CommissioningModeProvider.h

```diff
@@ -31,6 +32,9 @@ class DLL_EXPORT CommissioningModeProvider
 {
 public:
     virtual Dnssd::CommissioningMode GetCommissioningMode() const = 0;
+#if CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC
+    virtual bool IsJointFabricEnabled() = 0;
```

Comment:
Oh, and for the separate PR that will add this: having conditionally defined bits of vtables is a good recipe for mismatched vtable layouts and security bugs.  Should generally be avoided.

Separately, this new virtual method needs documentation.

Reply:
Thanks for the feedback. Based on your recommendation, this PR has been updated to focus exclusively on implementing mDNS advertisement. A follow-up PR will handle the OJCW functionalities and address the comments regarding unit tests. Closing out this thread for now.